src/main/scala/cromwell/engine/backend/local/LocalBackend.scala

```diff
@@ -135,16 +133,19 @@ class LocalBackend extends Backend with LazyLogging {
    */
   private def stageWorkflowInputs(descriptor: WorkflowDescriptor): HostInputs = {
     val hostInputsPath = Paths.get(hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
-    descriptor.actualInputs map {stageInput(_, hostInputsPath)}
+    descriptor.actualInputs.map {case(name, value) => stageInput(name, value, hostInputsPath)}
   }
 
-  private def stageInput(nameAndValue: (String, WdlValue), hostInputsPath: Path): (String, WdlValue) = {
-    val (name, value) = nameAndValue
-    val hostPathAdjustedValue = value match {
-      case w: WdlFile => stageWdlFile(w, hostInputsPath)
-      case x => x
-    }
-    name -> hostPathAdjustedValue
+  private def stageInputArray(array: WdlArray, hostInputsPath: Path): WdlArray = array.map(stageWdlValue(_, hostInputsPath))
+
+  private def stageWdlValue(value: WdlValue, hostInputsPath: Path) = value match {
+    case w:WdlFile => stageWdlFile(w, hostInputsPath)
+    case a:WdlArray => stageInputArray(a, hostInputsPath)
+    case x => x
+  }
+
+  private def stageInput(name: String, value: WdlValue, hostInputsPath: Path): (String, WdlValue) = {
```

Comment:
That'd be even better. The use of that particular pairing might not be as widespread as I'm picturing, I just know I saw it in my JES travels. I wouldn't worry about it, more of a thing to think about - if we're seeing that pair track together repeatedly it's probably A Thing.